fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/aggregate/AggregateAccumulators.java

```diff
@@ -0,0 +1,108 @@
+
+import javax.annotation.Nonnull;
+
+/**
+ * A class that contains static initializers for various aggregate accumulators.
+ */
+public class AggregateAccumulators {
+
+    // ------------------ SUM Aggregators ------------------------------
+
+    public static RecordValueAccumulator<Integer, Integer> sumInteger(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new IntegerState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    public static RecordValueAccumulator<Long, Long> sumLong(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new LongState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    public static RecordValueAccumulator<Float, Float> sumFloat(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new FloatState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    public static RecordValueAccumulator<Double, Double> sumDouble(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new DoubleState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    // ------------------ MIN Aggregators ------------------------------
+
+    public static RecordValueAccumulator<Integer, Integer> minInteger(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new IntegerState(PrimitiveAccumulatorOperation.MIN));
+    }
+
+    public static RecordValueAccumulator<Long, Long> minLong(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new LongState(PrimitiveAccumulatorOperation.MIN));
+    }
+
+    public static RecordValueAccumulator<Float, Float> minFloat(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new FloatState(PrimitiveAccumulatorOperation.MIN));
+    }
+
+    public static RecordValueAccumulator<Double, Double> minDouble(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new DoubleState(PrimitiveAccumulatorOperation.MIN));
+    }
+
+    // ------------------ MAX Aggregators ------------------------------
+
+    public static RecordValueAccumulator<Integer, Integer> maxInteger(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new IntegerState(PrimitiveAccumulatorOperation.MAX));
+    }
+
+    public static RecordValueAccumulator<Long, Long> maxLong(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new LongState(PrimitiveAccumulatorOperation.MAX));
+    }
+
+    public static RecordValueAccumulator<Float, Float> maxFloat(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new FloatState(PrimitiveAccumulatorOperation.MAX));
+    }
+
+    public static RecordValueAccumulator<Double, Double> maxDouble(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new DoubleState(PrimitiveAccumulatorOperation.MAX));
+    }
+
+    // ------------------ AVERAGE Aggregators ------------------------------
```

Comment:
I think it may be better to only have aggregators that obey some rules for intermediate rollups. Which means average would be done by dividing two aggregators at the end, rather than as a first-class on of its own.

Reply:
As discussed in the meeting, we would like to commit the current implementation and add this feature as a future enhancement.

Reply:
As discussed this should be postponed to later.